### PR TITLE
[RW-4626][risk=no] Puppeteer: Fix Registration tests

### PR DESCRIPTION
**Pull request by aweng98**
Updated tests to work with new Create Account UI and removed non-working code.
Unrelated to broken test, I also saw some flaky behavior while running tests repeatedly. I refactored few lines of code that should remove flaky behavior.

### CODE REVIEW

File: e2e/tests/user/registration.spec.ts [hunk 1]
```diff
@@ -54,16 +55,22 @@ describe('User registration tests:', () => {
 
     await createAccountPage.acceptTermsOfUseAgreement();
     let nextButton = await createAccountPage.getNextButton();
+    await nextButton.waitUntilEnabled();
```
Comment (NehaBroad) [inline e2e/tests/user/registration.spec.ts]: Can we add comments here to // Step 2: Terms of service?

File: e2e/tests/user/registration-ui.spec.ts [hunk 2]
```diff
@@ -133,35 +135,47 @@ describe('User registration tests:', () => {
...
+    expect(await nextButton.isCursorNotAllowed()).toEqual(true);
 
-    // the NEXT button on User Information page should be disabled until all required fields are filled
-    const userInforPageButton = await createAccountPage.getNextButton();
-    const cursor = await userInforPageButton.isCursorNotAllowed();
-    expect(cursor).toEqual(true);
+    const institutionSelect = new SelectComponent(page, 'Select your institution');
+    await institutionSelect.select(INSTITUTION_VALUE.VANDERBILT);
+
+    const emailAddress = await Textbox.forLabel(page, {textContains: FIELD_LABEL.INSTITUTION_EMAIL, ancestorNodeLevel: 2});
```
Comment (NehaBroad) [inline e2e/tests/user/registration-ui.spec.ts]: Do we add negative test too? Like invalid email etc

File: e2e/app/aou-elements/select-component.ts [hunk 3]
```diff
@@ -1,30 +1,35 @@
...
   }
 
-  async select(textValue: string) {
+  async select(textValue: string): Promise<string> {
     await this.open(2); // with 2 retries
-    const selector = this.componentXpath() + `//li[@class='p-dropdown-item'][normalize-space(.)="${textValue}"]`;
+    const selector = this.dropdownXpath() + `//li[contains(normalize-space(text()), "${textValue}")]`;
     const selectValue = await this.page.waitForXPath(selector, { visible: true });
+    const selectElement = new BaseElement(this.page, selectValue);
+    const str = await selectElement.getTextContent();
```
Comment (NehaBroad) [inline e2e/app/aou-elements/select-component.ts]: super nit: change variable name from str to textContent

File: e2e/app/base-page.ts [hunk 4]
```diff
@@ -221,4 +222,19 @@ export default abstract class BasePage {
...
+   */
+  async takeScreenshot(fileName: string) {
+    const dir = 'logs/screenshot';
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir);
+    }
+    const timestamp = new Date().getTime();
+    const screenshotFile = `${dir}/${fileName}_${timestamp}.png`;
+    await this.page.screenshot({path: screenshotFile, fullPage: true});
+    console.log('screenshot taken: ' + screenshotFile);
```
Comment (NehaBroad) [inline e2e/app/base-page.ts]: nice!

File: e2e/app/create-account-page.ts [hunk 5]
```diff
@@ -129,27 +127,37 @@ export default class CreateAccountPage extends BasePage {
...
   async selectYearOfBirth(year: string) {
-    const dropdown = new SelectComponent(this.page, FIELD_LABEL.YEAR_OF_BIRTH);
+    const dropdown = new SelectComponent(this.page, FIELD_LABEL.YEAR_OF_BIRTH, 2);
     await dropdown.select(year);
   }
 
   // Combined steps to make test code cleaner and shorter
 
-  // Step 1: Enter Invitation key
+  // Step 1: Fill out Institution
```
Comment (NehaBroad) [inline e2e/app/create-account-page.ts]: nit: Fill out institution details - to clarify its just not institution name but email and role as well

File: e2e/app/google-login.ts [hunk 6]
```diff
@@ -89,8 +89,13 @@ export default class GoogleLoginPage extends BasePage {
   /**
    * Open All-of-Us Google login page.
    */
-  async goto(): Promise<void> {
-    await this.page.goto(configs.uiBaseUrl + configs.loginUrlPath, {waitUntil: ['networkidle0', 'domcontentloaded'], timeout: 0});
+  async load(): Promise<void> {
+    const url = configs.uiBaseUrl + configs.loginUrlPath;
+    await this.page.goto(url, {waitUntil: ['networkidle0', 'domcontentloaded'], timeout: 30000}).catch((err) => {
+      console.error('Google login page not found. ' + err);
+      this.takeScreenshot('GoogleLoginNotFound');
```
Comment (NehaBroad) [inline e2e/app/google-login.ts]: May not be part of this PR but it will be nice to have a util method to handle errors where in we can just pass the flow name say invitation key -> it will print error and take a screen shot with file name invitation key

File: e2e/app/aou-elements/select-component.ts (hunk 3 above)
Comment (aweng98) [reply e2e/app/aou-elements/select-component.ts]: 👍

File: e2e/app/create-account-page.ts (hunk 5 above)
Comment (aweng98) [reply e2e/app/create-account-page.ts]: 👍

File: e2e/app/google-login.ts (hunk 6 above)
Comment (aweng98) [reply e2e/app/google-login.ts]: I'll make a note of this, for next PR.

File: e2e/tests/user/registration-ui.spec.ts (hunk 2 above)
Comment (aweng98) [reply e2e/tests/user/registration-ui.spec.ts]: Good idea. I think that would be a new test in a new PR. purpose of this PR is mainly for fixing broken tests.

File: e2e/tests/user/registration.spec.ts (hunk 1 above)
Comment (aweng98) [reply e2e/tests/user/registration.spec.ts]: has a comment, it is at line #51.
